src/write_to_file.cc

```diff
@@ -47,20 +51,22 @@ int writeValueToFile(
 }
 
 
-void writeVectorToFile(std::vector<double> v, std::string pFilename){
+void writeVectorToFile(std::vector<double> v, fs::path filename){
```

Comment:
Some other option would be to call filename something like filenameSkeleton and create the full filename from this skeleton.